### PR TITLE
update change log entry types and add to pull request template

### PR DESCRIPTION
**Pull request by zacharyburnett**
# Description

# Checklist:

- [x] pre-commit checks ran successfully
- [ ] tests ran successfully
- [ ] for a public change, added news fragment (`changes/<PR#>.<changetype>.rst`)
- [ ] for a public change, updated documentation
- [ ] for any new features, unit tests were added

### CODE REVIEW

**Comment by zacharyburnett** [comment]
- [ ] pre-commit checks ran successfully
- [ ] tests ran successfully
- [ ] for a public change, added news fragment (`changes/<PR#>.<changetype>.rst`)
- [ ] for a public change, added <details><summary>news fragment (`changes/<PR#>.<changetype>.rst`)</summary>

    - ``changes/<PR#>.feature.rst``: new feature
    - ``changes/<PR#>.bugfix.rst``: bug fix
    - ``changes/<PR#>.doc.rst``: documentation change
    - ``changes/<PR#>.removal.rst``: deprecation or removal of public API
    - ``changes/<PR#>.misc.rst``: not of interest to users; use an empty file
  </details>
- [ ] for a public change, updated documentation
- [ ] for any new features, unit tests were added

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -29,6 +29,13 @@ If this PR fixes an issue, please add closing keywords (eg 'fixes #XXX')
 
 - [ ] pre-commit checks ran successfully
 - [ ] tests ran successfully
-- [ ] for a public change, added news fragment (`changes/<PR#>.<changetype>.rst`)
+- [ ] for a public change, added <details><summary>news fragment (`changes/<PR#>.<changetype>.rst`)</summary>
```
Comment (braingram) [inline .github/pull_request_template.md:32]:
```suggestion
- [ ] for a public change, added a [towncrier](https://towncrier.readthedocs.io/en/stable/) <details><summary>news fragment (`changes/<PR#>.<changetype>.rst`)</summary>
```

**Comment by braingram** [comment]
Changes look good and it's good to see the changelog check fail due to the `misc` fragment. Would you rename `1828.misc.rst` it to `1828.general.rst`?

**Review by braingram — APPROVED** [review]
Thanks!

**Pull request merged**